spec/models/group_spec.rb

```diff
@@ -13,15 +13,40 @@
     it { is_expected.to validate_presence_of(:leader) }
     it { is_expected.to belong_to(:draw) }
     it { is_expected.to validate_presence_of(:draw) }
-    it { is_expected.to have_many(:members) }
+    it { is_expected.to have_many(:memberships) }
+    it { is_expected.to have_many(:members).through(:memberships) }
   end
 
-  describe '#name' do
-    it "includes the leader's name" do
-      leader = instance_spy('User', name: 'Name')
-      group = FactoryGirl.build_stubbed(:group)
-      allow(group).to receive(:leader).and_return(leader)
-      expect(group.name).to include(leader.name)
+  describe 'size validations' do
+    it 'must be available in the draw' do
+      group = FactoryGirl.build(:group)
+      allow(group.draw).to receive(:suite_sizes).and_return([1])
+      group.size = 2
+      expect(group.valid?).to be_falsey
+    end
+    it 'must not have more members than the size' do
```

Comment:
Hm, is this testing the group validations or the membership validations? I feel like we'd fail on the membership size since the group is full, so we'd never create the membership, therefore not update the counter cache, and therefore not hit the member count validation on the group. This is opening up a can of worms in my head - we can test this by creating an invalid group whose status is incorrect (thereby allowing the membership to be created), but then we end up with a borked db which sounds like a terrible idea.

I guess the membership count vs size validation is useful in ensuring that you can't reduce the size of a group below its memberships, so maybe that's what we should actually be testing? What do you think?